### DISCUSSION TITLE
Do we have plan to support java UDF

### DISCUSSION THREAD

**Discussion by HaoChen-ch** [facebookincubator/velox#10649] [Ideas]
Support for Java UDF is motivated by the following points:

1. the Java language is widely used, rich support for related library
functions, and a low threshold for development and debugging.
2. Some big data ecologies such as  Spark, Hive, etc. already exist a
large number of ready-made UDF, we need to be compatible with these udf to
reduce the migration costs of users.

So do we have plan to support java UDF in velox ?

**Comment by aditi-pandit** [comment]
Hi @HaoChen-ch : 

Thanks for your question. 

Velox has support for remote functions https://github.com/facebookincubator/velox/tree/main/velox/functions/remote. Would this help for your use-case ?

We have run some java functions using this internally at IBM.

@wills-feng

**Reply by HaoChen-ch** [reply]
Hi @aditi-pandit :

Thanks for the replay.

I will try to run some java functions using the remote functions.Hope this help for my use-case.
